internal/controller/instaslice_daemonset.go

```diff
@@ -211,96 +204,102 @@ func (r *InstaSliceDaemonsetReconciler) Reconcile(ctx context.Context, req ctrl.
 						log.FromContext(ctx).Error(shutdownErr, "error to perform nvml.Shutdown")
 					}
 				}()
-
-				availableGpus, ret := nvml.DeviceGetCount()
 				if ret != nvml.SUCCESS {
 					log.FromContext(ctx).Error(ret, "Unable to get device count")
 				}
 				//TODO: any GPU can fail creating CI and GI
 				// if simulator mode is on do not perform NVML calls
 				// TODO: move this logic to a new vendor specific file
-				if _, exists := cachedPreparedMig[allocations.PodName]; !exists {
+				if prep, exists := cachedPreparedMig[allocations.PodUUID]; !exists || isPreparedMigEmpty(prep) {
 					placement := nvml.GpuInstancePlacement{}
-					for i := 0; i < availableGpus; i++ {
+					device, retCodeForDevice := nvml.DeviceGetHandleByUUID(allocations.GPUUUID)
+					if retCodeForDevice != nvml.SUCCESS {
+						log.FromContext(ctx).Error(ret, "error getting GPU device handle")
+					}
+					var giProfileId, ciProfileId int
+					for _, item := range instaslice.Spec.Migplacement {
+						if item.Profile == profileName {
+							giProfileId = item.Giprofileid
+							ciProfileId = item.Giprofileid
+						}
+					}
+					giProfileInfo, retCodeForGi := device.GetGpuInstanceProfileInfo(giProfileId)
+					if retCodeForGi != nvml.SUCCESS {
+						log.FromContext(ctx).Error(retCodeForGi, "error getting GPU instance profile info", "giProfileInfo", giProfileInfo, "retCodeForGi", retCodeForGi)
```

Comment:
same as https://github.com/openshift/instaslice-operator/pull/141/files#r1793645627

Reply:
We log the error and see if we can use the cache or walk MIG devices to set prepared. this transient error indicates an issue with the vendor GPU call.